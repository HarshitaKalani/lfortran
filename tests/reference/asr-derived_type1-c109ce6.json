{
    "basename": "asr-derived_type1-c109ce6",
    "cmd": "lfortran --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/derived_type1.f90",
    "infile_hash": "ed39e2004e29457f0f83e6a1badf6ecbee98749ca624594fc0160e52",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-derived_type1-c109ce6.stdout",
    "stdout_hash": "2ff638d4c0a097d66da7953d7cc08de5453e2a2533dd87f97af661b5",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}