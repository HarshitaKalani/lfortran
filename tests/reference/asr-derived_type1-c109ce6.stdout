(TranslationUnit
    (SymbolTable
        1
        {
            derived_type1:
                (Module
                    (SymbolTable
                        2
                        {
                            1_enum_escape_newline:
                                (ExternalSymbol
                                    2
                                    1_enum_escape_newline
                                    3 newline
                                    enum_escape
                                    []
                                    newline
                                    Public
                                ),
                            enum_escape:
                                (StructType
                                    (SymbolTable
                                        3
                                        {
                                            achar:
                                                (ExternalSymbol
                                                    3
                                                    achar
                                                    5 achar
                                                    lfortran_intrinsic_builtin
                                                    []
                                                    achar
                                                    Private
                                                ),
                                            newline:
                                                (Variable
                                                    3
                                                    newline
                                                    [tfc]
                                                    Local
                                                    (FunctionCall
                                                        3 achar
                                                        ()
                                                        [((IntegerConstant 10 (Integer 4)))
                                                        ((Var 2 tfc))]
                                                        (Character 1 1 ())
                                                        (StringConstant
                                                            "\n"
                                                            (Character 1 1 ())
                                                        )
                                                        ()
                                                    )
                                                    (StringConstant
                                                        "\n"
                                                        (Character 1 1 ())
                                                    )
                                                    Save
                                                    (Character 1 1 ())
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    enum_escape
                                    []
                                    [newline]
                                    Source
                                    Public
                                    .false.
                                    .false.
                                    []
                                    ()
                                    ()
                                ),
                            nl:
                                (Variable
                                    2
                                    nl
                                    [toml_escape]
                                    Local
                                    (StringConcat
                                        (StringConstant
                                            " "
                                            (Character 1 1 ())
                                        )
                                        (StructInstanceMember
                                            (Var 2 toml_escape)
                                            2 1_enum_escape_newline
                                            (Character 1 1 ())
                                            (StringConstant
                                                "\n"
                                                (Character 1 1 ())
                                            )
                                        )
                                        (Character 1 2 ())
                                        (StringConstant
                                            " \n"
                                            (Character 1 2 ())
                                        )
                                    )
                                    (StringConstant
                                        " \n"
                                        (Character 1 2 ())
                                    )
                                    Parameter
                                    (Character 1 2 ())
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            tfc:
                                (Variable
                                    2
                                    tfc
                                    []
                                    Local
                                    (IntrinsicElementalFunction
                                        SelectedCharKind
                                        [(StringConstant
                                            "DEFAULT"
                                            (Character 1 7 ())
                                        )]
                                        0
                                        (Integer 4)
                                        (IntegerConstant 1 (Integer 4))
                                    )
                                    (IntegerConstant 1 (Integer 4))
                                    Parameter
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            toml_escape:
                                (Variable
                                    2
                                    toml_escape
                                    []
                                    Local
                                    (StructTypeConstructor
                                        2 enum_escape
                                        [(())]
                                        (Struct
                                            2 enum_escape
                                        )
                                        ()
                                    )
                                    ()
                                    Parameter
                                    (Struct
                                        2 enum_escape
                                    )
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                )
                        })
                    derived_type1
                    [lfortran_intrinsic_builtin]
                    .false.
                    .false.
                ),
            lfortran_intrinsic_builtin:
                (IntrinsicModule lfortran_intrinsic_builtin)
        })
    []
)
